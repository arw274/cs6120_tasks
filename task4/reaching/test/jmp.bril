# Block 0 reaching definitions:
# Block 1 reaching definitions:
# Block 2 reaching definitions: 
#   From block 0 : var a instr 0
#   From block 1 : var a instr 0

@main {
    a: int = const 0;
    jmp .end;
    a: int = const 1;
.end:
    ret;
}