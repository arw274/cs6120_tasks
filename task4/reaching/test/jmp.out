Function: main

Block 0: [{'dest': 'a', 'op': 'const', 'type': 'int', 'value': 0}, {'labels': ['end'], 'op': 'jmp'}]
Block 1: [{'dest': 'a', 'op': 'const', 'type': 'int', 'value': 1}]
Block 2: [{'label': 'end'}, {'op': 'ret'}]
Block 0 reaching definitions:
Block 1 reaching definitions:
Block end reaching definitions:
  From block 1 : var a instr 0
  From block 0 : var a instr 0

